examples/plot_metric_examples.py

```diff
@@ -0,0 +1,410 @@
+"""
+Algorithms walkthrough
+~~~~~~~~~~~~~~~~~~~~~~
+
+This is a small walkthrough which illustrates all the Metric Learning
+algorithms implemented in metric-learn, and also does a quick
+visualisation which can help understand which algorithm might be best
+suited for you.
+
+Of course, depending on the data set and the constraints your results
+will look very different; you can just follow this and change your data
+and constraints accordingly.
+
+"""
+
+# License: BSD 3 clause
+# Authors: Bhargav Srinivasa Desikan <bhargavvader@gmail.com>
+
+######################################################################
+# Imports
+# ^^^^^^^
+#
+
+from sklearn.manifold import TSNE
+from sklearn.utils import shuffle
+
+import metric_learn
+import numpy as np
+from sklearn.datasets import make_classification
+
+# visualisation imports
+import matplotlib.pyplot as plt
+np.random.seed(42)
+
+
+######################################################################
+# Loading our data-set and setting up plotting
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# 
+# We will be using the IRIS data-set to illustrate the plotting. You can
+# read more about the IRIS data-set here:
+# `link <https://en.wikipedia.org/wiki/Iris_flower_data_set>`__.
+
+X, Y = make_classification(n_classes=3, n_clusters_per_class=2,
+                           n_informative=3, class_sep=4., n_features=5,
+                           n_redundant=0, shuffle=True)
+
```

Comment:
maybe add `n_samples=100` explicitly

and add explanation about what the dataset looks like:
- 100 points in 3 classes with 2 clusters per class
- 5 features, among which 3 are informative (correlated with the class labels) and two are random noise with large magnitude

Reply:
I agree, done